electrical/breakbeam.py

```diff
@@ -0,0 +1,47 @@
+import RPi.GPIO as GPIO
+import time
+
+class Breakbeam:
+    ACCEPTABLE_TIME = 50 # in seconds
+    BEAM_PINS = {17:'50%_1', 22:'50%_2', 23:'100%_1', 24:'100%_2'}
+
+    def __init__(self):
+        for BEAM_PIN in Breakbeam.BEAM_PINS.keys():
+            GPIO.setmode(GPIO.BCM)
+            GPIO.setup(BEAM_PIN, GPIO.IN, pull_up_down=GPIO.PUD_UP)
+            GPIO.add_event_detect(BEAM_PIN, GPIO.BOTH, callback=self.break_beam_callback)
+            print(self.timer(BEAM_PIN))
+        self.beam_broken = False
+        self.blocked_sensors = []
+    
+    def break_beam_callback(self, channel):
+        print(not GPIO.input(channel), channel)
+        self.beam_broken = not GPIO.input(channel)
+        if not GPIO.input(channel):
+            self.timer(channel)
```

Comment:
```suggestion
        self.beam_broken = not GPIO.input(channel)
        if self.beam_broken:
            # Bream beak detected interruption, run timer check to see whether it is a false reading
            self.timer(channel)
```

Reply:
^^^^^^

Reply:
if you make a variable, try to use it as much as u can! (ie use self.beam_broken instead of writing not GPIO.input(channel) again)